src/components/Table/BatchActionTable.tsx

```diff
@@ -2,48 +2,56 @@ import { useEffect, useRef, useState } from 'react';
 import exampleData from './example-data.json';
 
 const BatchActionTable = () => {
-  const table = useRef<HTMLTdsTableBodyElement>(null);
+  const batchActionTable = useRef<HTMLTdsTableElement>(null);
   const modal = useRef<HTMLTdsModalElement>(null);
-  const [data, setData] = useState<string>('[]');
+  const [selectedData, setSelectedData] = useState<any[]>();
+  const [allSelected, setAllSelected] = useState(exampleData.every((row) => row.selected));
+  const [data, setData] = useState(exampleData);
 
   useEffect(() => {
-    if (table.current) {
-      table.current.bodyData = exampleData;
-    }
-  });
+    const handleSelectAll = (event: any) => {
+      setAllSelected(true);
+      const updatedData = data?.map((row) => ({
+        ...row,
+        selected: event.detail.checked,
+      }));
+      setData(updatedData);
+    };
+    batchActionTable.current?.addEventListener('tdsSelectAll', handleSelectAll);
+    return () => {
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      batchActionTable.current?.removeEventListener('tdsSelectAll', handleSelectAll);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Select all doesn't work. 
Appears to be because event.detail.checked should be event.detail.allSelected (although I think i'd prefer 'checked' or 'value')

BTW i think we should add data to the dependency array and remove the comments, to encourage less error prone code

Reply:
Feel free to add a commit and fix it.

Reply:
To my understanding the empty dependency array is a totally legitimate pattern. At least for our use case, we want to add an event listener only one (on initial render) and remove it when the component is removed. I get that eslist doesn't like it, but that's a rule we can control.

Reply:
Ignore previous comment